components/Slider/tests/unit/SliderTests.m

```diff
@@ -913,6 +913,80 @@ - (void)testThumbElevationDefault {
   XCTAssertEqual(self.slider.thumbElevation, MDCShadowElevationNone);
 }
 
+- (void)testThumbRadiusForState {
+  // Given
+  self.slider.statefulAPIEnabled = YES;
+  CGFloat normal = 12.f;
+  CGFloat disabled = 10.f;
+  CGFloat highlighted = 8.f;
+  CGFloat selected = 16.f;
+
+  // When
+  [self.slider setThumbRadius:normal forState:UIControlStateNormal];
+  [self.slider setThumbRadius:disabled forState:UIControlStateDisabled];
+  [self.slider setThumbRadius:highlighted forState:UIControlStateHighlighted];
+  [self.slider setThumbRadius:selected forState:UIControlStateSelected];
+
+  // Then
+  self.slider.enabled = YES;
+  self.slider.selected = NO;
+  self.slider.highlighted = NO;
+  XCTAssertEqual(self.slider.thumbRadius, normal);
+
+  self.slider.enabled = NO;
+  self.slider.selected = NO;
+  self.slider.highlighted = NO;
+  XCTAssertEqual(self.slider.thumbRadius, disabled);
+
+  self.slider.enabled = YES;
+  self.slider.selected = NO;
+  self.slider.highlighted = YES;
+  XCTAssertEqual(self.slider.thumbRadius, highlighted);
+
+  self.slider.enabled = YES;
+  self.slider.selected = YES;
+  self.slider.highlighted = NO;
+  XCTAssertEqual(self.slider.thumbRadius, selected);
+}
+
+- (void)testThumbRadiusHasNoEffectWhenStatefulAPIIsEnabled {
```

Comment:
If I turn on stateful API, set the radius, then turn off the stateful API, should it be ignored? For the reverse (stateful off, setRadius, stateful on), updating the radius will affect the stateful API.

Reply:
No, if you set the radius when stateful API is on then that is the thumb radius. If you turn it off later how would we know what it was before setting it? We would have to store a value inside the implementation `nonStatefulThumbRadius` we don't do that for any other properties and that seems *hacky*.